react-carousel/my-app/src/Carousel.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  FaChevronCircleLeft,
+  FaChevronCircleRight,
+  FaCircle,
+  FaRegCircle,
+} from 'react-icons/fa';
+import { Buttons } from './Buttons';
+import { ImageDisplay } from './ImageDisplay';
+import { useEffect, useState } from 'react';
+
+export function Carousel() {
+  const [imageIndex, setImageIndex] = useState(0);
+  const images = [
+    {
+      src: '/images/fushiguro.webp',
+      alt: 'Megumi Fushiguro',
+    },
+    {
+      src: '/images/inumaki.webp',
+      alt: 'Toge Inumaki',
+    },
+    {
+      src: '/images/itadori.webp',
+      alt: 'Yuji Itadori',
+    },
+    {
+      src: '/images/kugisaki.webp',
+      alt: 'Nobara Kugisaki',
+    },
+    {
+      src: '/images/panda.webp',
+      alt: 'Panda',
+    },
+    {
+      src: '/images/zen-in.webp',
+      alt: "Maki Zen'in",
+    },
+  ];
+  // const callback = useCallback(() => imageIndex, []);
+  useEffect(() => {
+    const intervalId = setTimeout(
+      () => setImageIndex((prev) => (prev >= images.length - 1 ? 0 : prev + 1)),
+      1000
+    );
+    return () => clearTimeout(intervalId);
+  }, [imageIndex]);
```

Comment:
This code doesn't depend directly on `imageIndex`. So it shouldn't really be in the dependency array. But you need it here to make the timer restart when the image rotates. There are 2 ways to address this:
* Change the code on line 42 to use `imageIndex` instead of using the `prev` approach
* Get useCallback working (see below)